### PR TITLE
Optimize IndexCoord scheduler loop

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
/kind improvement

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/indexcoord/index_builder.go [hunk 1]
```diff
@@ -169,14 +169,17 @@ func (ib *indexBuilder) run() {
...
-		ib.process(buildID)
+		if ib.process(buildID) {
+			log.Ctx(ib.ctx).Debug("there is no IndexNode available or etcd is not serviceable, wait a minute...")
+			break
+		}
 	}
 }
 
-func (ib *indexBuilder) process(buildID UniqueID) {
+func (ib *indexBuilder) process(buildID UniqueID) bool {
```
Comment (czs007) [inline internal/indexcoord/index_builder.go:182]: flip the return value

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**